scripts/mount-sd.sh

```diff
@@ -39,39 +52,60 @@ if [ "$ACTION" = "add" ]; then
         MINER_STATUS="$(dbus-send --type=method_call --print-reply --session --dest=org.freedesktop.Tracker1.Miner.Files /org/freedesktop/Tracker1/Miner/Files org.freedesktop.Tracker1.Miner.GetStatus | grep -o 'Idle')"
         STORE_STATUS="$(dbus-send --type=method_call --print-reply --session --dest=org.freedesktop.Tracker1 /org/freedesktop/Tracker1/Status org.freedesktop.Tracker1.Status.GetStatus | grep -o 'Idle')"
         test "$MINER_STATUS" = "Idle" -a "$STORE_STATUS" = "Idle" && break
-        systemd-cat -t mount-sd /bin/echo "Waiting $count seconds for tracker"
+        log "Waiting $count seconds for tracker"
         sleep $count ; 
         count=$(( count + count ))
     done
 
-    test -d $MNT/${UUID} || mkdir -p $MNT/${UUID}
-    chown $DEF_UID:$DEF_GID $MNT $MNT/${UUID}
-    touch $MNT/${UUID}
+    test -d ${TARGET} || mkdir -p ${TARGET}
+    chown $DEF_UID:$DEF_GID $MNT ${TARGET}
+    touch ${TARGET}
 
     case "${TYPE}" in
 	vfat|exfat)
-	    mount ${DEVNAME} $MNT/${UUID} -o uid=$DEF_UID,gid=$DEF_GID,$MOUNT_OPTS,utf8,flush,discard || /bin/rmdir $MNT/${UUID}
+        if mount ${DEVNAME} ${TARGET} -o uid=$DEF_UID,gid=$DEF_GID,$MOUNT_OPTS,utf8,flush,discard; then
+	        ln -s "${TARGET}" "$(get_path_fname ${UUID})"
+        else
+            /bin/rmdir ${TARGET}
+        fi
 	    ;;
 	# NTFS support has not been tested but it's being left to please the ego of an engineer!
 	ntfs)
-	    mount ${DEVNAME} $MNT/${UUID} -o uid=$DEF_UID,gid=$DEF_GID,$MOUNT_OPTS,utf8 || /bin/rmdir $MNT/${UUID}
+	    mount ${DEVNAME} ${TARGET} -o uid=$DEF_UID,gid=$DEF_GID,$MOUNT_OPTS,utf8 || /bin/rmdir ${TARGET}
 	    ;;
 	*)
-	    mount ${DEVNAME} $MNT/${UUID} -o $MOUNT_OPTS || /bin/rmdir $MNT/${UUID}
+        if mount ${DEVNAME} ${TARGET} -o $MOUNT_OPTS; then
+            COMMON_DIR=${TARGET}/NemoMobileData
+            if ! [ -d "${COMMON_DIR}" ]; then
+                log "Creating common directory ${COMMON_DIR}"
+                if mkdir ${COMMON_DIR} && chmod 1777 ${COMMON_DIR}; then
+                    ln -s "${COMMON_DIR}" "$(get_path_fname ${UUID})"
+                fi
+            elif ! $(su nemo -c "test -w ${COMMON_DIR}"); then
```

Comment:
This puts in the assertion that the system user is `nemo`, and that there's only one user.